src/openzaak/components/autorisaties/models.py

```diff
@@ -0,0 +1,152 @@
+        called as part of `transaction.on_commit`.
+        """
+        from .utils import send_applicatie_changed_notification
+
+        qs = cls.objects.select_related("applicatie").prefetch_related(
+            "applicatie__autorisaties"
+        )
+
+        to_delete = []
+        to_keep = []
+        to_add = []
+
+        for spec in qs:
+            existing_autorisaties = [
+                autorisatie
+                for autorisatie in spec.applicatie.autorisaties.all()
+                if autorisatie.component == spec.component
+            ]
+
+            for autorisatie in existing_autorisaties:
+                # schedule for deletion if existing objects differ from the spec
+                if (
+                    autorisatie.max_vertrouwelijkheidaanduiding
+                    != spec.max_vertrouwelijkheidaanduiding
+                ):
+                    to_delete.append(autorisatie)
+                    continue
+
+                if set(autorisatie.scopes) != set(spec.scopes):
+                    to_delete.append(autorisatie)
+                    continue
+
+                to_keep.append(autorisatie)
+
+            TypeModel = apps.get_model(COMPONENT_TO_MODEL[spec.component])
+            field = COMPONENT_TO_FIELD[spec.component]
+
+            for obj in TypeModel.objects.all():
+                url = build_absolute_url(obj.get_absolute_api_url())
+
+                autorisatie = Autorisatie(
+                    applicatie=spec.applicatie,
+                    component=spec.component,
+                    scopes=spec.scopes,
+                    max_vertrouwelijkheidaanduiding=spec.max_vertrouwelijkheidaanduiding,
+                    **{field: url}
+                )
+                to_add.append(autorisatie)
+
+        Autorisatie.objects.filter(pk__in=[autorisatie.pk for autorisatie in to_delete])
+
+        # de-duplicate - whatever is in to_keep should not be added again
+        existing_urls = defaultdict(list)
+        for autorisatie in to_keep:
+            if autorisatie.component not in COMPONENT_TO_FIELD:
+                continue
+            url = getattr(autorisatie, COMPONENT_TO_FIELD[autorisatie.component])
+            existing_urls[autorisatie.component].append(url)
+
+        _to_add = []
+        for autorisatie in to_add:
+            if autorisatie.component not in COMPONENT_TO_FIELD:
+                continue
+            url = getattr(autorisatie, COMPONENT_TO_FIELD[autorisatie.component])
+            if url in existing_urls[autorisatie.component]:
+                continue
+            _to_add.append(autorisatie)
+
+        # created the de-duplicated, missing autorisaties
+        Autorisatie.objects.bulk_create(_to_add)
+
+        # determine which notifications to send
+        changed = {autorisatie.applicatie for autorisatie in (to_delete + _to_add)}
+        for applicatie in changed:
+            send_applicatie_changed_notification(applicatie)
```

Comment:
Oh, so notifications are supported?

In that case the TODO [here](https://github.com/open-zaak/open-zaak/pull/307/files#diff-6cad5efe4cac826fd10c4c55bb505adaR219) can be removed and #355 can be closed again.